pydantic_xml/serializers/factories/homogeneous.py

```diff
@@ -17,6 +18,81 @@ class HomogeneousSerializerFactory:
     Homogeneous collection type serializer factory.
     """
 
+    class TextSerializer(Serializer):
+        def __init__(
+                self, model: Type['pxml.BaseXmlModel'], model_field: pd.fields.ModelField, ctx: Serializer.Context,
+        ):
+            assert model_field.sub_fields and len(model_field.sub_fields) == 1
+            if (
+                isclass(model_field.type_) and issubclass(model_field.type_, pxml.BaseXmlModel) or
+                issubclass(model_field.type_, tuple)
+            ):
+                raise errors.ModelFieldError(
+                     model.__name__, model_field.name, "Inline list value should be of scalar type",
+                )
+
+        def serialize(
+                self, element: XmlElementWriter, value: Collection[Any], *, encoder: XmlEncoder,
+                skip_empty: bool = False,
+        ) -> Optional[XmlElementWriter]:
+            if value is None or skip_empty and len(value) == 0:
+                return element
+
+            encoded = " ".join(encoder.encode(val) for val in value)
+            element.set_text(encoded)
+            return element
+
+        def deserialize(self, element: Optional[XmlElementReader]) -> Optional[List[Any]]:
+            if element is None:
+                return None
+
+            text = element.pop_text()
+
+            if text is None:
+                return []
```

Comment:
I think it is better to return `None` in this case otherwise default values will not work. For example for this model:

```python
class RootModel(BaseXmlModel, tag="model"):
    values: Tuple[int, ...] = (0, 0)
```

if `deserialize` returns `[]` that code will not work because pydantic uses default values only if a value is not provided. But if `deserialize` returns `[]` it will not be filtered out by the model deserializer and the default value will not be set which is not expected behavior I suppose.

Reply:
I don't know the best way to resolve multiple conversations on github but I've updated the PR with fixes to your comments.

I also realized recently that I probably should have created an issue first to discuss design and then made a PR. I apologize that I skipped that order and will do better next time. Thank you for your patience in dealing with me :)